src/Hazelcast.Net/Serialization/SerializationService.ReadWrite.cs

```diff
@@ -210,7 +210,20 @@ public async ValueTask<T> ToObjectAsync<T>(object dataObj, ToObjectState state)
         /// <exception cref="HazelcastException">Internal error.</exception>
         public async ValueTask<object> ToObjectAsync(object dataObj, ToObjectState state)
         {
-            throw new NotSupportedException("We don't have a compact serializer for now.");
+            var schemaId = state.SchemaId;
+            var safety = 1000;
```

Comment:
I think it's too much try. It looks like pooling to me. Why don't we fail more fast?

Reply:
This is the idea: this method gets an object to deserialize + the ID of a schema that we know is missing. We fetch the schema (and throw if not found), then we try again to deserialize the object. This may fail if *another* schema is missing, in which case we loop. Eventually we'll have all schemas that we need. Except if there's some kind of a bug in our code, and we end up looping forever trying to somehow fetch the same missing schema over and over again. That should of course never happen. This safety thing here exists "just in case" to make sure the loop absolutely cannot become infinite. Let's say it's me not trusting me :) The number needs to be greater than the maximum number of schemas that we might need to fetch when deserializing. 1000 sounded big enough. I mean, 10 would be too small, we *could* need to fetch 10 schemas.

So now... if you look at our code globally, this could be replaced by `while (true)` but I am always scared of these things. Maybe I can add a comment to explain what I just wrote above?